WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -0,0 +1,170 @@
+      }
+    });
+    return subRegionsMap;
+  }, [data]);
+
+  if (loading || saving) return <Loading />;
+  if (error || saveError) return <Errored />;
+
+  return (
+    <>
+      <Header as="h2">Region Manager</Header>
+      <List bulleted>
+        {regions.map((region) => (
+          <List.Item key={region.id}>
+            <List.Content>
+              <UserGroupVisibility userGroup={region} save={save} sync={sync} />
+              {region.name}
+              <List.List>
+                {subRegions[region.id]?.map((subRegion) => (
+                  <List.Item key={subRegion.id}>
+                    <List.Content>
+                      <UserGroupVisibility userGroup={subRegion} save={save} sync={sync} />
+                      {subRegion.name}
+                    </List.Content>
+                  </List.Item>
+                ))}
+              </List.List>
+            </List.Content>
+          </List.Item>
+        ))}
+      </List>
+      <ButtonGroup>
+        <Button
+          onClick={() => setOpenModalType('newRegion')}
+          disabled={saving}
+        >
+          Add new region
+        </Button>
+        <Button
+          onClick={() => setOpenModalType('newSubregion')}
+          disabled={saving}
+        >
+          Add new subregion
+        </Button>
+      </ButtonGroup>
+      <Modal
+        size="fullscreen"
+        onClose={() => {
+          closeModal();
+          setNewRegion(defaultRegion);
+        }}
+        open={openModalType === 'newRegion' || openModalType === 'newSubregion'}
+      >
+        <Modal.Content>
+          <Form>
+            {openModalType === 'newSubregion' && (
+              <Form.Dropdown
+                label="Parent region"
+                fluid
+                selection
+                value={newRegion.parent_group_id}
+                onChange={(e, { value }) => setNewRegion({ ...newRegion, parent_group_id: value })}
+                options={regions.map((region) => ({
+                  key: region.id,
+                  text: region.name,
+                  value: region.id,
+                }))}
+              />
+            )}
+            <Form.Input
+              label="Name"
+              value={newRegion.name}
+              onChange={(e, { value }) => setNewRegion({ ...newRegion, name: value })}
+            />
+            <Form.Button
+              onClick={() => {
+                closeModal();
+                setNewRegion(defaultRegion);
+              }}
+            >
+              Cancel
+            </Form.Button>
+            <Form.Button
+              disabled={
+                newRegion.name.length === 0
+                || (openModalType === 'newSubregion' && !newRegion.parent_group_id)
+              }
+              onClick={() => {
+                closeModal();
+                save(addUserGroupsUrl, newRegion, () => sync(), { method: 'POST' }, setSaveError);
+                sync();
```

Comment:
Why are you calling `sync` in the `onSuccess` hook of the `save` method _and_ after the `save` invocation?

Reply:
Seems like you are syncing twice

Reply:
My bad, removed one of them. :-)